ClemBot.Bot/bot/services/tag_service.py

```diff
@@ -21,9 +21,11 @@ def __init__(self, *, bot):
     @BaseService.Listener(Events.on_guild_message_received)
     async def on_guild_message_received(self, message: discord.Message) -> None:
 
+        tagprefix = await self.bot.custom_tag_prefix_route.get_custom_tag_prefixes(message.guild.id, raise_on_error=True)
+        tagprefix = tagprefix[0]
         tags_content = []
         tag_found = False
-        pattern = re.compile(f'(^|\\s)[{TAG_PREFIX}](\\w+)')
+        pattern = re.compile(f'(^|\\s)[{tagprefix}](\\w+)')
```

Comment:
There is no benefit to using `re.compile(...)` here because the pattern is being re-compiled every time a message is received. I suggest compiling a pattern for the default tag prefix and storing it as an instance variable and for custom tag prefixes maybe just use `re.findall("pattern", message.content)`.